pandas/_typing.py

```diff
@@ -26,8 +26,11 @@
 )
 ArrayLike = TypeVar("ArrayLike", ABCExtensionArray, np.ndarray)
 DatetimeLikeScalar = TypeVar("DatetimeLikeScalar", Period, Timestamp, Timedelta)
-Dtype = Union[str, np.dtype, ExtensionDtype]
+Dtype = Union[str, np.dtype, "dtypes.ExtensionDtype"]
```

Comment:
If I don't do it like this, I get a  circular import error.

Reply:
Does this actually resolve back to "ExtensionDtype" or does it replace it with `Any`?

Reply:
Yes, it works in PyCharm + I tried setting the string to a random string, which mypy won't accept.

Reply:
Good to know. Wondering if we can extend that to solve #27146 (separate PR obviously)

Reply:
Yeah, I think that should be possible, though it wold be even better if the ABCs could be used...

Reply:
can't you use 'ExtendionDtype' here?

Reply:
?

Reply:
hmm I guess maybe revert to what you had before is more clear (e.g. do't use the string)